pkg/trustengine/trust_engine.go

```diff
@@ -0,0 +1,62 @@
+package trustengine
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+type trustEngineUtils interface {
+	*piperhttp.Client
+}
+
+type TrustEngineSecret struct {
+	Token string `json:"sonar,omitempty"`
+}
+
+type TrustEngineConfiguration struct {
+	ServerURL string
+	Token     string
+}
+
+func GetTrustEngineSecret(refName string, client *piperhttp.Client, trustEngineConfiguration TrustEngineConfiguration) (string, error) {
```

Comment:
What about handing in a list of systems and returning a list of tokens here already to be prepared?

Reply:
Done